cmd/prepare-storage.go

```diff
@@ -239,7 +235,16 @@ func retryFormattingDisks(firstDisk bool, endpoints []*url.URL, storageDisks []S
 					// Validate formats load before proceeding forward.
 					err := genericFormatCheck(formatConfigs, sErrs)
 					if err == nil {
-						printHealMsg(firstEndpoint.String(), storageDisks, printOnceFn())
+						// Healing should always be trigerred from first endpoint.
+						firstEndpoint := endpoints[0]
+						if firstEndpoint == nil {
+							return errInvalidArgument
+						}
+						// We do not need the path component, just extract Host and Scheme.
+						printHealMsg(&url.URL{
+							Host:   firstEndpoint.Host,
+							Scheme: firstEndpoint.Scheme,
+						}, storageDisks, printOnceFn())
```

Comment:
for singlenode-XL endpoint.Path will be relevant but here path is not getting passed